src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -971,6 +973,29 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
         history.push(url, { app: this.props.app })
     }
 
+    @OF.autobind
+    async onClickReplayAll() {
+        this.setState({
+            isReplayAllInProgress: true
+        })
+
+        for (const trainDialog of this.props.trainDialogs) {
```

Comment:
Seems like it should just replay the train dialogs that need replay rather than all of them.

Reply:
Yes, what is reliable way to determine that state?  I assumed checking for invalid state was not enough, because I thought it was possible that any dialog that needs replay for other reason that being invalid.

There was similar proposal in the description:

> Determine if any dialogs have unsaved changes and only allow replay all if this is the case

If we can determine if dialogs need replay we can also do this.

Reply:
I should add, that if we do this change which we would want to do for performance we can't label the button "Replay All" because it's not guaranteed to be "All" dialogs anymore.  I'm not sure of a good name. We can't say "Replay Invalid" either.

Perhaps we have to do the work for selection and add "Replay Selected"